workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/HandleCreatePlayerRequestSystem.cs

```diff
@@ -12,83 +11,85 @@ namespace Improbable.Gdk.PlayerLifecycle
-                            createPlayerRequest = request
-                        }
-                    ));
-                }
+                commandSystem.SendCommand(entityRequest);
             }
+        }
 
-            for (var i = 0; i < entityCreationResponseData.Length; ++i)
+        private void HandlePlayerCreateEntityResponses()
+        {
+            var responses = commandSystem.GetResponses<WorldCommands.CreateEntity.ReceivedResponse>();
+            for (var i = 0; i < responses.Count; i++)
             {
-                var entityCreationResponses = entityCreationResponseData.CreateEntityResponses[i];
-                var responder = entityCreationResponseData.CreatePlayerResponders[i];
-
-                foreach (var receivedResponse in entityCreationResponses.Responses)
+                var response = responses[i];
+                if (!(response.Context is PlayerCreationRequestContext requestContext))
                 {
-                    if (!(receivedResponse.Context is PlayerCreationRequestContext requestContext))
-                    {
-                        // Ignore non-player entity creation requests
-                        continue;
-                    }
-
-                    if (receivedResponse.StatusCode != StatusCode.Success || !receivedResponse.EntityId.HasValue)
-                    {
-                        responder.ResponsesToSend.Add(new PlayerCreator.CreatePlayer
-                            .Response(requestContext.createPlayerRequest.RequestId,
-                                $"Failed to create player: \"{receivedResponse.Message}\""));
+                    // Ignore non-player entity creation requests
+                    continue;
+                }
 
-                        continue;
-                    }
+                if (response.StatusCode != StatusCode.Success)
+                {
+                    var responseFailed = new PlayerCreator.CreatePlayer.Response(
+                        requestContext.createPlayerRequest.RequestId,
+                        $"Failed to create player: \"{response.Message}\""
+                    );
 
-                    responder.ResponsesToSend.Add(new PlayerCreator.CreatePlayer
-                        .Response(requestContext.createPlayerRequest.RequestId, new CreatePlayerResponseType
+                    commandSystem.SendResponse(responseFailed);
+                }
+                else
+                {
+                    var responseSuccess = new PlayerCreator.CreatePlayer.Response(
+                        requestContext.createPlayerRequest.RequestId,
+                        new CreatePlayerResponseType
                         {
-                            CreatedEntityId = receivedResponse.EntityId.Value
-                        }));
+                            CreatedEntityId = response.EntityId.Value
+                        }
+                    );
+
+                    commandSystem.SendResponse(responseSuccess);
```

Comment:
Nit: This can be refactored to:

```
var requestId = requestContext.createPlayerRequest.RequestId;

var response = response.StatusCode == StatusCode.Success 
	? new PlayerCreator.CreatePlayer(requestId, new CreatePlayerResponseType { CreatedEntityId = response.EntityId.Value })
	: new PlayerCreator.CreatePlayer(requestId, $"Failed to create player: \"{response.Message}\"");

commandSystem.SendResponse(response);
```

Reply:
Thats way too big for a conditional operator

Reply:
I disagree, but its fine either way